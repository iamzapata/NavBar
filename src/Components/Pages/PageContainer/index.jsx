import React, { PureComponent } from "react"
import { node } from "prop-types"
import "./PageContainer.scss"

class PageContainer extends PureComponent {
  constructor(props) {
    super(props)
    this.DOMnode = React.createRef()
  }
  componentDidMount() {
    const el = this.DOMnode.current
    el.style.opacity = 0
    el.style.visibility = "hidden"
    setTimeout(() => {
      el.style.transition = "opacity 1300ms ease-in-out"
      el.style.opacity = 1
      el.style.visibility = "visible"
    }, 0)
  }
  render() {
    const { children } = this.props
    return (
      <div ref={this.DOMnode} className="PageContainer container">
        {children}
      </div>
    )
  }
}

PageContainer.propTypes = {
  children: node.isRequired
}

export default PageContainer
