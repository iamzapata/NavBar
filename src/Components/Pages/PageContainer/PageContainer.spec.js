import React from "react"
import { render, cleanup } from "react-testing-library"
import PageContainer from "./index"

describe("<PageContainer />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(<PageContainer />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
