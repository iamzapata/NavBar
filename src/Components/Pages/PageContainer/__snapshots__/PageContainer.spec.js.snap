// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> Should render self 1`] = `
<div
  class="PageContainer container"
  style="opacity: 0; visibility: hidden;"
>
  <div />
</div>
`;
